src/legacy/index.ts

```diff
@@ -314,8 +323,10 @@ async function buildSubtree(
         addLabel(depTree, 'pruned', 'true');
       }
       return depTree;
+    } else {
+      // No double insertions, save some time
```

Comment:
this comment is clear as part of this PR, but can become confusing by itself, so 👍 on the change itself, but maybe remove the comment